SEED-platform/seed
# Remove coparent check for merges

#### Any background context you want to provide?
See the issue for more detailed background.

There was a check on the -States involved in merges before the merges were actually executed. The tax lot version of this check was inadvertently checking the PropertyState model. 

Upon further research, the check ensures that a -State does not have a coparent before merging.

If a -State _does_ have a coparent , it was previously part of a merge. If a -State was previously part of a merge, it is no longer associated with a -View (a new -State is used for the resulting, merged state).

Given that, it shouldn't really be "targetable" for future merges, so my thinking was that this was not a worthwhile check in the first place. 

**After writing this out, I can envision an API user potentially targeting a -State not associated to a -View (bypassing the logic above). It's unconventional and likely shouldn't be allowed - maybe validating the need for this check. So I could see an argument for keeping this check in place (and just correcting it).  Thoughts?**

#### What's this PR do?
Rather than "correct" the inadvertent reference to PropertyState models, this removes the check entirely for both Property merges and Tax Lot merges. 

#### How should this be manually tested?
Merging should work just as before - so this should be monkey tested.

The biggest part of this review is chiming in on whether we want to remove this check or keep it in (and fix it for Tax Lots of course).

#### What are the relevant tickets?
#1975 

#### Screenshots (if appropriate)
